### PR TITLE
Refactor/238 parent pom property

### LINKED ISSUE #238: Refactor PomFileValidator to maven projects with partent

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #238

### CODE REVIEW

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileIO.java [hunk 1]
```diff
@@ -66,18 +66,17 @@ private void writePomFile(final Document document, final OutputStream outputStre
         }
     }
 
-    /**
-     * Read a pom file as XML document.
-     * 
-     * @param pomFile pom file
-     * @return xml document
+    /*
+     * Inspired by https://stackoverflow.com/a/33564346
```
Comment (kaklakariada) [inline project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileIO.java:70]: This is public now. Maybe add a real comment.
```suggestion
     * Trim the whitespace from the given node and its children.
     *
     * Inspired by https://stackoverflow.com/a/33564346
```

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileGeneratorTest.java [hunk 2]
```diff
@@ -20,12 +20,28 @@
...
 import com.exasol.projectkeeper.ProjectKeeperModule;
+import com.exasol.projectkeeper.config.ProjectKeeperConfig;
 
 class PomFileGeneratorTest {
 
+    static Stream<Arguments> testGenerationWithParentPomCases() {
+        return Stream.of(//
+                Arguments.of(null, "../pom.xml" /* that seems to be a default in the reader */),
+                Arguments.of("./myParentPom.xml", "./myParentPom.xml")//
+        );
```
Comment (kaklakariada) [inline project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileGeneratorTest.java:31]: When using only simple types (strings, int, ...) you could use the `@CsvSource` annotation and specify the parameters directly at the test.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileGeneratorTest.java (hunk 2 above)
Comment (jakobbraun) [reply project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileGeneratorTest.java:31]: I had that in the beginning but the null was not very clear in that notation

Comment (kaklakariada) [reply project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileGeneratorTest.java:31]: 👍

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**